### PR TITLE
add expand-collapse button

### PR DESCRIPTION
**Pull request by madhur-tandon**
Closes https://github.com/jupyterlab-contrib/search-replace/issues/27

### CODE REVIEW

**Comment by madhur-tandon** [comment]
Not sure why the test fails, it passes locally for me

**Review by fcollonval — CHANGES_REQUESTED** [review]

File: src/searchReplace.tsx [hunk 1]
```diff
@@ -294,6 +294,23 @@ const SearchReplaceElement = (props: IProps) => {
...
+              document.getElementsByClassName('search-tree-files');
+            for (let i = 0; i < elements.length; i++) {
+              const treeItem = elements[i];
+              if (treeItem.hasAttribute('expanded')) {
+                treeItem.removeAttribute('expanded');
+              } else {
+                treeItem.setAttribute('expanded', '');
+              }
+            }
+          }}
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply src/searchReplace.tsx]: done

File: src/searchReplace.tsx [hunk 2]
```diff
@@ -294,6 +294,23 @@ const SearchReplaceElement = (props: IProps) => {
...
+        >
+          <expandAllIcon.react></expandAllIcon.react>
```
[the remark this answers is not in the document]
Comment (madhur-tandon) [reply src/searchReplace.tsx]: done

File: ui-tests/tests/search.spec.ts [hunk 3]
```diff
@@ -172,3 +172,26 @@ test('should make a new request on refresh', async ({ page }) => {
...
+  await Promise.all([
+    page.waitForResponse(
+      response =>
+        /.*search\/\?query=strange/.test(response.url()) &&
+        response.request().method() === 'GET'
+    ),
+    page.locator('input[type="search"]').press('Enter'),
+    page.locator('[title="button to expand and collapse all results"]').click()
+  ]);
+
```
Comment (fcollonval) [inline ui-tests/tests/search.spec.ts:192]: You are doing too much action at once.

You should first start the query in `Promise.all`. Probably waiting for 

```ts
  await page.waitForSelector('.jp-search-replace-tab >> .jp-progress', {
    state: 'hidden'
  })
```

After it to ensure you get the results displayed.

Then only click on the expand/collapse button.

Currently you are doing something no human can do: validating the input box and clicking on the button at the same time. (The click actually has no effect as there are no results.)

File: src/searchReplace.tsx [hunk 4]
```diff
@@ -294,6 +312,21 @@ const SearchReplaceElement = (props: IProps) => {
         >
           <refreshIcon.react></refreshIcon.react>
         </Button>
+        <Button
```
Comment (fcollonval) [inline src/searchReplace.tsx:315]: You should `disabled` the button if there is no results.

To disable, add the property `disabled={condition}` to the button.

**Review by fcollonval — CHANGES_REQUESTED** [review]

File: ui-tests/tests/search.spec.ts (hunk 3 above)
Comment (madhur-tandon) [reply ui-tests/tests/search.spec.ts:194]: fixed

**Review by fcollonval — APPROVED** [review]
Thanks @madhur-tandon

**Pull request merged**